lib/client.js

```diff
@@ -1,11 +1,38 @@
 var PROMISE_TIMEOUT = 5000, // 5 seconds
     ZAF_EVENT       = /^zaf\./,
     version         = require('version'),
+    Utils           = require('utils'),
     Promise         = window.Promise || require('../vendor/native-promise-only'),
     when            = require('utils').when,
     pendingPromises = {},
     ids             = {};
 
+// @internal
+// #timeoutReject(name, params)
+//
+// Reject a request if required, given the rejection function, name (get, set or invoke)
+// and arguments to that request. Falls back to 5000 second timeout with few exceptions.
+//
+function timeoutReject(reject, name, paramsArray) {
+  switch(name) {
+  case 'invoke': {
+    var allowed = ['instances.create'];
+    if (allowed.length !== paramsArray.length) {
+      break;
+    }
+    var match = paramsArray.every(function(value, index) {
+      return value === allowed[index];
+    });
+    if (match) {
+      return NaN; // timer IDs are numbers; embrace the JavaScript 😬
+    }
+  }
+  }
+  return setTimeout(function() {
```

Comment:
So if you make a bulk `invoke` (not supported at the moment) with both allowed and not allowed APIs it can timeout?